pinning-services/index.js

```diff
@@ -210,26 +275,73 @@ function updateDnsRecordDigitalOcean(domain, recordType, recordName, recordData,
   console.log(`GET ${urlGet}`);
   return fetch(urlGet, { headers: headers })
     .then(res => res.json())
-    .then(json => json['domain_records'].filter(record => record['type'] === recordType && record['name'] === recordName))
+    .then(json => {
+      if (json['id'] && json['id'] === 'not_found') {
+        return addDomainAccountDigitalOcean(domain, doToken)
+          .then(res => {
+            if (res) {
+              return [];
+            } else {
+              throw `Could not create domain ${domain}`;
+            }
+          });
+      } else {
+        return json['domain_records'].filter(record => record['type'] === recordType && record['name'] === recordName);
+      }
+    })
     .then(txt => {
+      let isDelete = false;
+      let isUpdate = false;
+
+      // If there is anything other then one entry found, 
+      // all entries should be removed then one entry added
+      if (txt.length != 1)
+        isUpdate = true
+
+      // If there is a record, and recordData is empty an string, delete only
+      if (txt.length === 1 && recordData === '') {
+        isDelete = true;
+        isUpdate = false
+      }
+
+      // If there is a record, and the data in that record differs from recordData, update
+      if (txt.length === 1 && txt[0]['data'] != recordData)
+        isUpdate = true;
+
+      // If there is no record, and the recordData is an empty string, don't do anything
+      if (txt.length === 0 && recordData === '') {
+        isUpdate = false;
+        isDelete = false;
+      }
+
+      // If there is an update, there needs to be a delete first
+      if (isUpdate === true) isDelete = true;
```

Comment:
This line is equivalent to:
```suggestion
      isDelete = isUpdate;
```

So everything `isDelete` above is useless.

Reply:
Collapsed